plugins/services/src/js/reducers/serviceForm/JSONReducers/__tests__/MultiContainerEnvironmentVariables-test.ts

```diff
@@ -8,15 +8,10 @@ describe("Environment Variables", () => {
     });
 
     it("returns an array of transactions", () => {
+      const environment = { FOO: "value" };
       expect(
-        MultiContainerEnvironmentVariables.JSONParser({
-          environment: { FOO: "value" },
-        })
-      ).toEqual([
-        { type: ADD_ITEM, value: 0, path: ["env"] },
-        { type: SET, value: "FOO", path: ["env", 0, "key"] },
-        { type: SET, value: "value", path: ["env", 0, "value"] },
-      ]);
+        MultiContainerEnvironmentVariables.JSONParser({ environment })
+      ).toEqual([{ type: SET, value: environment, path: ["env"] }]);
     });
```

Comment:
@janisz this change is the gist of it all:

```typescript
// before
[
  new Transaction(["env"], 0, ADD_ITEM),
  new Transaction(["env.0.key"], "FOO", SET),
  new Transaction(["env.0.value"], "value", SET)
  new Transaction(["env"], 1, ADD_ITEM),
  new Transaction(["env.1.key"], "bar", SET),
  new Transaction(["env.1.value"], "val", SET)
]

// after 
[
  new Transaction(["env"], { FOO: "value", bar: "val" }, SET)
]
```

we can do that because `KVForm` abstracts the use case where you'd specify a key twice. imagine you want to write a couple env vars in the form:
```
user         | peter
user_surname | pan
```

right when you type the second key of `user_surname`, the form would "swallow" the `user` as javascript can not handle duplicate keys on objects. this is why all of this `.key` and `.value` complexity has been built in the first place.